instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/NeverSuppressOrStoreStrategy.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.api.instrumenter;
+
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.SpanKind;
+import io.opentelemetry.context.Context;
+
+final class NeverSuppressOrStoreStrategy extends SpanSuppressionStrategy {
```

Comment:
With the above naming suggestion, `NoopSpanSuppresionStrategy` seems ok for this one.